java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -1117,28 +1117,39 @@ public boolean existsAndIsAccessible(final String path) {
             head(path);
         } catch (MantaClientHttpResponseException e) {
-                if (e.getServerCode().equals(MantaErrorCode.DIRECTORY_DOES_NOT_EXIST_ERROR)) {
+            switch (e.getServerCode()) {
+                case DIRECTORY_DOES_NOT_EXIST_ERROR:
                     LOG.error("{} Directory invalid for given path: {}", e.getMessage(), path);
-                } else if (e.getServerCode().equals(MantaErrorCode.ACCOUNT_DOES_NOT_EXIST_ERROR)) {
+                    break;
+                case ACCOUNT_DOES_NOT_EXIST_ERROR:
                     LOG.error("{} Non-existent account for given path: {}", e.getMessage(), path);
-                } else if (e.getServerCode().equals(MantaErrorCode.AUTHORIZATION_FAILED_ERROR)
-                    || e.getServerCode().equals(MantaErrorCode.INVALID_CREDENTIALS_ERROR)) {
+                    break;
+                case AUTHORIZATION_FAILED_ERROR:
+                case INVALID_CREDENTIALS_ERROR:
                     LOG.error("{} Invalid authorization credentials for given path: {}", e.getMessage(), path);
-                } else if (e.getServerCode().equals(MantaErrorCode.RESOURCE_NOT_FOUND_ERROR)) {
+                    break;
+                case RESOURCE_NOT_FOUND_ERROR:
                     LOG.error("{} Unavailable Resource for given path: {}", e.getMessage(), path);
-                } else if (e.getServerCode().equals(MantaErrorCode.INVALID_KEY_ID_ERROR)) {
+                    break;
+                case INVALID_KEY_ID_ERROR:
                     LOG.error("{} Invalid key-id for given path: {}", e.getMessage(), path);
-                } else if (e.getServerCode().equals(MantaErrorCode.JOB_NOT_FOUND_ERROR)) {
+                    break;
+                case JOB_NOT_FOUND_ERROR:
                     LOG.error("{} Non-existent Job Id for given path: {}", e.getMessage(), path);
-                } else if (e.getServerCode().equals(MantaErrorCode.MULTIPART_UPLOAD_STATE_ERROR)) {
+                    break;
+                case MULTIPART_UPLOAD_STATE_ERROR:
                     LOG.error("{} Unavailable Resource for given path: {}", e.getMessage(), path);
-                } else if (e.getServerCode().equals(MantaErrorCode.REQUEST_TIMEOUT_ERROR)) {
+                    break;
+                case REQUEST_TIMEOUT_ERROR:
                     LOG.error("{} Request time-out for given path: {}", e.getMessage(), path);
-                } else if (e.getServerCode().equals(MantaErrorCode.INTERNAL_ERROR)) {
+                    break;
+                case INTERNAL_ERROR:
                     LOG.error("{} Internal Server error for given path: {}", e.getMessage(), path);
-                } else if (e.getServerCode().equals(MantaErrorCode.SNAPLINKS_DISABLED_ERROR)) {
+                    break;
+                case SNAPLINKS_DISABLED_ERROR:
                     LOG.error("{} Snaplinks disabled for given path: {}", e.getMessage(), path);
-                }
+                    break;
```

Comment:
Maybe add a default case that logs a generic message?

Reply:
Done.